bessctl/pcapngdump.py

```diff
@@ -67,18 +67,19 @@ def read_exactly(f, size):
     Read from file f exactly size bytes.
     Might return a buffer smaller than size if EOF or error.
     """
-    ret = b''
+    ret = []
     while True:
-        ret += f.read(size)
-        size = size - len(ret)
+        buf = f.read(size)
+        ret.append(buf)
+        size = size - len(buf)
         if size == 0:
             break
```

Comment:
Hm, reading this again, what happens at hard EOF? `f.read` returns 0, so `size` does not decrement.  Needs an `if buf == b'': break` to handle that. Then you can change the loop to be `while size:` and get rid of the second test/break.

Reply:
You're right, I wasn't handling it correctly.  There was another big problem that prevented EOF from happening: the fifo was set in non blocking mode.  For low packet rates this meant that `f.read` raised EAGAIN immediately, ending the capture before it could even start.

Both these problems should now be fixed, thanks!